src/main/java/io/reactivesocket/util/ReactiveSocketProxy.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.reactivesocket.util;
+
+import io.reactivesocket.Payload;
+import io.reactivesocket.ReactiveSocket;
+import io.reactivesocket.rx.Completable;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+
+/**
+ * Wrapper/Proxy for a ReactiveSocket.
+ * This is useful when we want to override a specific method.
+ */
+public class ReactiveSocketProxy implements ReactiveSocket {
+    protected final ReactiveSocket child;
+    private final Function<Subscriber<? super Payload>, Subscriber<? super Payload>> subscriberWrapper;
+
+    public ReactiveSocketProxy(ReactiveSocket child, Function<Subscriber<? super Payload>, Subscriber<? super Payload>> subscriberWrapper) {
```

Comment:
You're right, it's most of the time a cleaner approach.
That been said, this is pretty convenient to be able to do that here.